### PR TITLE
ICU W3 warning fixes

### PR DESCRIPTION
**Pull request by daniel-ju**

### CODE REVIEW

File: icu4c/source/common/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (jefgen) [inline icu4c/source/common/.vscode/settings.json]: Definitely want to exclude this file from your pull-request. :)

(I think we might want to exclude the ".vscode" folder using the .gitignore file in the future. Don't worry about that now though, that can be a totally separate PR later on)

File: icu4c/source/common/bytestriebuilder.cpp [hunk 2]
```diff
@@ -485,10 +485,10 @@ BytesTrieBuilder::writeDeltaTo(int32_t jumpTarget) {
             length=2;
         } else {
             if(i<=0xffffff) {
-                intBytes[0]=(char)BytesTrie::kFourByteDeltaLead;
+                intBytes[0]=0xfeu;
```
Comment (jefgen) [inline icu4c/source/common/bytestriebuilder.cpp]: Hmm. I'm not sure we want to go from using defined constants to using magic numbers.
What happens when somebody changes kFourByteDeltaLead in the future? This line is now broken.

File: icu4c/source/common/common.vcxproj.filters [hunk 3]
```diff
@@ -777,30 +777,15 @@
     <ClInclude Include="umapfile.h">
       <Filter>data &amp; memory</Filter>
     </ClInclude>
-    <CustomBuild Include="unicode\locdspnm.h">
```
Comment (jefgen) [inline icu4c/source/common/common.vcxproj.filters]: Please revert these changes. :)

File: icu4c/source/common/dictbe.cpp [hunk 4]
```diff
@@ -325,7 +325,7 @@ ThaiBreakEngine::divideUpDictionaryRange( UText *text,
                         // two characters after uc were not 0x0E4C THANTHAKHAT before
                         // checking the dictionary. That is just a performance filter,
                         // but it's not clear it's faster than checking the trie.
-                        int32_t candidates = words[(wordsFound + 1) % THAI_LOOKAHEAD].candidates(text, fDictionary, rangeEnd);
+                        candidates = words[(wordsFound + 1) % THAI_LOOKAHEAD].candidates(text, fDictionary, rangeEnd);
```
Comment (jefgen) [inline icu4c/source/common/dictbe.cpp]: What was the warning here? Going from a small scope to a large scope seems possibly risky. (Might be fine but I don't have the context)

File: icu4c/source/common/rbbi.cpp [hunk 5]
```diff
@@ -848,7 +848,7 @@ int32_t RuleBasedBreakIterator::handleNext() {
 
        #ifdef RBBI_DEBUG
             if (gTrace) {
-                RBBIDebugPrintf("             %4ld   ", utext_getNativeIndex(&fText));
```
Comment (jefgen) [inline icu4c/source/common/rbbi.cpp]: What was the warning here? I think the %41d was just for the white-space padding.

File: icu4c/source/common/dictbe.cpp (hunk 4 above)
Comment (daniel-ju) [reply icu4c/source/common/dictbe.cpp]: The warning was "warning C4456: declaration of 'candidates' hides previous local declaration", but you're right I should be changing the variable name instead of changing the scope.

File: icu4c/source/common/rbbi.cpp (hunk 5 above)
Comment (daniel-ju) [reply icu4c/source/common/rbbi.cpp]: The warning here was "warning C4477: 'printf' : format string '%4ld' requires an argument of type 'long', but variadic argument 1 has type 'int64_t'", which is why I changed it to %lld.

Comment (jefgen) [reply icu4c/source/common/rbbi.cpp]: Oops, I thought that it was a "one" and didn't realize it was a "l". Thanks!

File: icu4c/source/common/locdspnm.cpp [hunk 6]
```diff
@@ -560,21 +560,21 @@ LocaleDisplayNamesImpl::adjustForUsageAndContext(CapContextUsage usage,
 }
 
 UnicodeString&
-LocaleDisplayNamesImpl::localeDisplayName(const Locale& locale,
+LocaleDisplayNamesImpl::localeDisplayName(const Locale& loc,
```
[the remark this answers is not in the document]
Comment (daniel-ju) [reply icu4c/source/common/locdspnm.cpp]: There was a warning due to "locale" being a class member and this declaration would hide the class member.

[the remark this answers is not in the document]
Comment (jefgen) [reply icu4c/source/common/locdspnm.cpp]: Ah, thanks!

File: icu4c/source/i18n/currpinf.cpp [hunk 7]
```diff
@@ -298,12 +298,12 @@ CurrencyPluralInfo::setupCurrencyPluralPattern(const Locale& loc, UErrorCode& st
         const char* pluralCount;
         while ((pluralCount = keywords->next(NULL, ec)) != NULL) {
             if ( U_SUCCESS(ec) ) {
-                int32_t ptnLen;
+                int32_t ptnLength;
```
Comment (jefgen) [inline icu4c/source/i18n/currpinf.cpp]: I think some of these changes might conflict with some white-space changes that I have in another PR, but that should be fine. It will be a good test for handling merge conflicts. :)

**Comment by jefgen** [comment]
Thanks for all this work @daniel-ju ! 

I'm going to close this Pull-Request into my **fork**, since there is now an open Pull-Request to get these changes upstream into the **official** ICU repo itself.  ( See https://github.com/unicode-org/icu/pull/18 ).
